File: src/main/java/com/mymedia/nagasu/dto/ImageUploadDto.java
package com.mymedia.nagasu.dto;

import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public record ImageUploadDto(
        String title,
        String artist,
        List<String> tags,
        String description,
        List<MultipartFile> images,
        MultipartFile thumbnail
) {
    public ImageUploadDto(String title, String artist, List<String> tags, String description,
                          List<MultipartFile> images) {
        this(title, artist, tags, description, images, null);
    }

    public ImageUploadDto(String title, String artist, String description,
                          List<MultipartFile> images) {
        this(title, artist, List.of(), description, images, null);
    }
}

File: src/main/kotlin/com/mymedia/streamer/dto/ImageUploadDto.kt
package com.mymedia.streamer.dto

import org.springframework.web.multipart.MultipartFile

data class ImageUploadDto(
    val title: String,
    val artist: String,
    val tags: List<String> = emptyList(),
    val description: String?,
    val images: List<MultipartFile>,       // @RequestPart("images") 역할
    val thumbnail: MultipartFile? = null   // @RequestPart("thumbnail") 역할
)